### PR TITLE
serialization: new annotation to rename an attribute in the serialization format

### PR DESCRIPTION
**Pull request by xymus**
By default, an attribute is identified in the serialization format by its Nit name. The `serialize_as` attribute changes this behavior and sets the name of an attribute in the serialization format. It works with both the JSON and the binary implementations.

This annotation can be useful to change the name of an attribute to what is expected by a remote service. Or to use identifiers in the serialization format that are reserved keywords in Nit (like `class` and `type`).

### CODE REVIEW

File: src/frontend/serialization_phase.nit [hunk 1]
```diff
@@ -56,6 +61,34 @@ redef class ADefinition
...
+
+		if not node isa AAttrPropdef then
+			toolcontext.error(node.location,
+				"Syntax Error: annotation `{text}` applies only to attributes.")
+			return
+		end
+
+		# Can't use `arg_as_string` because it needs the literal phase
+		var args = nat.n_args
+		if args.length != 1 or not args.first isa AStringFormExpr then
```
Comment (privat) [inline src/frontend/serialization_phase.nit]: use `arg_as_string`. see https://github.com/nitlang/nit/blob/master/src/annotation.nit#L42

**Comment by privat** [comment]
missing nitce :p

**Comment by privat** [comment]
+1 anyway

File: src/frontend/serialization_phase.nit (hunk 1 above)
Comment (xymus) [inline src/frontend/serialization_phase.nit]: It tried that, bu if I'm not mistaken `arg_as_string` needs the literal phase. This new phase creates a lot of literals so it runs before the literal phase.

Comment (privat) [inline src/frontend/serialization_phase.nit]: the literal phase was maybe a bad idea :(

**Comment by xymus** [comment]
Added the missing sav file for nitce.

**Pull request merged**